automation/prepare-and-start.sh

```diff
@@ -25,6 +25,21 @@ echo "[INFO] Starting docker."
 docker daemon 2> /dev/null &
 wait_docker
 
+# Authenticate with Balena registry if required
+BALENAOS_ACCOUNT="balena_os"
+if [ "$DEPLOY_TO" = "staging" ]; then
+	if [ -n "${BALENAOS_STAGING_TOKEN}" ]; then
+		echo "[INFO] Logging into $DEPLOY_TO as ${BALENAOS_ACCOUNT}"
+		export BALENARC_BALENA_URL=balena-staging.com
+		balena login --token ${BALENAOS_STAGING_TOKEN}
+	fi
+else
+	if [ -n "${BALENAOS_PRODUCTION_TOKEN}" ]; then
+		echo "[INFO] Logging into $DEPLOY_TO as ${BALENAOS_ACCOUNT}"
+		balena login --token ${BALENAOS_PRODUCTION_TOKEN}
+	fi
+fi
```

Comment:
This seems too complicated, can't we just have a single `BALENAOS_TOKEN` and overwrite the URL when `"$DEPLOY_TO" = "staging"`? Or is this something we have today and need to be compatible with it?

Reply:
@mtoman you're right that it can be simplified. 
```
# Authenticate with Balena registry if required                                    
BALENAOS_ACCOUNT="balena_os"                                                       
BALENAOS_TOKEN=${BALENAOS_PRODUCTION_TOKEN}                                        
if [ "$DEPLOY_TO" = "staging" ]; then                                              
         BALENAOS_TOKEN=${BALENAOS_STAGING_TOKEN}                                   
         export BALENARC_BALENA_URL=balena-staging.com                              
fi                                                                                 
if [ -n "${BALENAOS_TOKEN}" ]; then                                                
         echo "[INFO] Logging into $DEPLOY_TO as ${BALENAOS_ACCOUNT}"               
        balena login --token ${BALENAOS_TOKEN}                                     
fi  
```
Moving forward we will be pulling blocks from the balena registry, so authenticating here is credentials are provided prepares us for that.

Reply:
Fixed, could you please re-review @mtoman ?